src/device-api/common.js

```diff
@@ -27,20 +28,12 @@ export async function doRestart(appId, force) {
 			const imageIds = _.map(app.services, 'imageId');
 			applicationManager.clearTargetVolatileForServices(imageIds);
 
-			const currentServices = app.services;
-			app.services = [];
-			return deviceState
-				.pausingApply(() =>
-					deviceState
-						.applyIntermediateTarget(currentState, { skipLock: true })
-						.then(function () {
-							app.services = currentServices;
-							return deviceState.applyIntermediateTarget(currentState, {
-								skipLock: true,
-							});
-						}),
-				)
-				.finally(() => deviceState.triggerApplyTarget());
+			return deviceState.pausingApply(async () => {
+				return Bluebird.each(app.services, async (service) => {
```

Comment:
This looks better @20k-ultra 

I'm still a bit curious if this is equivalent to the implementation with `applyIntermediateTarget`. Maybe @CameronDiver can shed a bit more light into that since I don't fully understand that yet.

My main concern was whether some race condition could occur, but I suppose that is the purpose of the `pausingApply`

Reply:
`applyIntermediateTarget` sounds just like another way to apply a target which means the SV figures out what is needed to reach the tagret and then executes those steps. The V2 builds a list of steps with the only step being restart from the [composition steps](https://github.com/balena-io/balena-supervisor/blob/f6ba69144b926accc279ee9d6172ba3266beef3e/src/compose/composition-steps.ts#L197) file.

Reply:
Yeah, I understand that, I was thinking mostly in the sense of the 'chesterton fence'. You replaced a fairly complex piece of code, with a two line fix that does the same. I'm wondering if there are any reasons the old piece of code was made that way. 

In any case I have tested the whole PR on my device and it appears to be working well. 

I don't have permissions to approve the PR, so you might want to wait for Cameron. I'm waiting on this PR to be merged to finish the work on PR #1529